artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java

```diff
@@ -2140,6 +2140,7 @@ private int iterQueue(final int flushLimit,
                                       QueueIterateAction messageAction) throws Exception {
       int count = 0;
       int txCount = 0;
+      Integer expectedHits = messageAction.expectedHits();
```

Comment:
Whats the purpose of this?

Reply:
QueueControl.expire with a messageID will only expire a single record, however the iteration will go until the end. it's sort of an optimization.